src/Microsoft.AspNetCore.Server.Kestrel.Core/Internal/Http/FrameOfT.cs

```diff
@@ -28,15 +28,22 @@ public Frame(IHttpApplication<TContext> application, FrameContext frameContext)
         /// The resulting Task from this loop is preserved in a field which is used when the server needs
         /// to drain and close all currently active connections.
         /// </summary>
-        public override async Task RequestProcessingAsync()
+        public override async Task ProcessRequestsAsync()
         {
+            var dataRecieved = false;
+
             try
             {
                 while (!_requestProcessingStopping)
                 {
                     TimeoutControl.SetTimeout(_keepAliveTicks, TimeoutAction.CloseConnection);
 
-                    InitializeHeaders();
+                    if (dataRecieved)
```

Comment:
If you look at the full diff for FrameOfT, you'll see that's exactly where Reset() used to be called (there and in Frame.Start()). It was still behind a `if (!_requestProcessing)` flag though with a big comment explaining how request information needs to be preserved in this case and AdaptedConnection.PrepareRequest() might throw.

The way Reset() was called was always gross. This just puts all the callers closer together in one file. It's also more efficient since it delays or elides the first call to Reset(). I understand disliking waiting for Input.ReadAsync() to complete in order to ensure the AdaptedConnections property is set. That's why I call it out so call it out in a comment, but it just makes sense to delay Reset until we actually start processing a request.

Another thing I'm considering to make this even cleaner is to move the calls to `new Frame(Request|Response)Headers()` before the first call Reset() in order to avoid the `if (FrameRequestHeaders == null)` check that's currently in the Reset() method of this PR.